pkg/events/queue/queue.go

```diff
@@ -0,0 +1,14 @@
+package queue
+
+import (
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type CacheConfig EventQueue
+
+type EventQueue interface {
+	String() string
+	Setup() error
```

Comment:
what if the user doesn't call Setup? I usually tend to not have functions forcing the user to keep a state of their struct. I am suggesting to remove the `Setup` function from the interface and instead all the initialization/setup operation to be in a `NewEventQueueMem` function.

Reply:
Yes, I thought about it as well and it is fine for me. I'll ask you to clone my repository and add fixes on top of it (I'm gone today) if you are ok with that.

Reply:
sure, I already have a branch that takes your commits and on top of them has some more that should address my comments: https://github.com/mtcherni95/tracee/tree/michael-default-caching
I can close this PR and open a PR with the branch above, as we agreed. Let me know if confirmed then I'll proceed.